app/Http/Controllers/Setup/Controller.php

```diff
@@ -50,37 +57,46 @@ public function database() {
 
     public function test_database() {
         // Attempt to connect to the database with the provided credentials
-        try {
-            $db = new \PDO('mysql:dbname='.request('db_name').';host='.request('db_host'),
-                request('db_username'), request('db_password'));
-            $success = true;
-        } catch(\PDOException $e) {
-            $success = false;
+        if(!empty(request('db_name'))) {
+            try {
+                $db = new \PDO('mysql:dbname='.request('db_name').';host='.request('db_host'),
+                    request('db_username'), request('db_password'));
+                $success = true;
+            } catch(\PDOException $e) {
+                $success = false;
+            }
+        } else {
+            try {
+                DB::RAW('SELECT VERSION();');
+                $success = true;
+            } catch(\Exception $e) {
+                die($e->getMessage());
+                $success = false;
+            }
         }
 
-        if(!$success) {
+        if($success) {
+            // Store the DB connection info in the session for later
+            if(!empty(request('db_name'))) {
```

Comment:
this should be each value, and then wrapped in a method. request should not be the only source, nor the check